### ISSUE TITLE
Improve consistency and guidance of themes element

### ISSUE THREAD

**Issue by rob-metalinkage** [opengeospatial/ogcapi-records#394]
The specification [example for themes](https://docs.ogc.org/DRAFTS/20-004.html#sc_keywords_and_themes)  uses an optional `url as per 
```json
{
          "id": "dobson",
          "url": "https://geo.woudc.org/codelists.xml#WOUDC_InstrumentCode_dobson"
        },
```

The [repo example]( https://github.com/opengeospatial/ogcapi-records/blob/master/core/examples/json/record.json#L61) 

however doesnt..

```json
{ "id": "dobson" }, { "id": "brewer" },
```

Why is this an issue - because in this case the scheme cannot be used to disambiguate the term (its not a "namespace" - its a reference to a file using an arbitrary - and in this case different technology - and an arbitrary data model.  There is also the case where it is a large schema and no file could be useful - such as GBIF species taxonomy that would change before you could download it...

I think the URL is a good solution but needs better guidance,  such as if the theme `id` is not a valid URI then the scheme should be a valid namespace or the `url` element must be present.

the `id` could be a valid uri using CURIES with prefixes - such as wmo-instrument:dobson - maybe the "scheme" element could be augmented with an optional "namespace" and "prefix" that could be used to let clients know how to resolve.  Note that a custom JSON-LD context could also be provided to allow clients to resolve CURIES to unambiguous URLs.

Proposed guidance would be perhaps:

"In order to allow clients to unambiguously identify keywords without negotiating access, without further description available, to some arbitrary resolver for the scheme then the `url` property SHOULD be present for each theme entry."

or, if we want to make Records more flexible then we could state:

"If the `scheme` element is not a valid URL namespace than can be concatenated with theme `id` values then the property `namespace` should be present to support identification and possible resolution by dereferencing of theme `id` values. "

or even we could use a resolver using the link template pattern - this would allow either a service or a namespace pattern to be enforced.

At the vary least two things are needed:

1. improvements as suggested or some discussion is required about the burden imposed on systems to either enforce a profile with predicatable behaviour, or force clients to negotiate and manage schema/id pairs as identifiers. 
2. Make repo examples consistent with inline examples in specification

**Comment by pvretano** [comment]
28-OCT-2024: Assigned to @tomkralidis since he is most familiar with the content.

**Comment by tomkralidis** [comment]
Addressed in #395.
